alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/implementations/actions/SendScafiMessage.scala

```diff
@@ -11,14 +11,14 @@ package it.unibo.alchemist.model.implementations.actions
 
 import it.unibo.alchemist.model.ScafiIncarnationUtils
 import it.unibo.alchemist.model.ScafiIncarnationUtils._
-import it.unibo.alchemist.model.implementations.nodes.ScafiNode
 import it.unibo.alchemist.model.interfaces._
+
 import java.util.stream.Collectors
 import scala.jdk.CollectionConverters._
 
 class SendScafiMessage[T, P<:Position[P]](
   env: Environment[T, P],
-  node: ScafiNode[T, P],
+  node: Node[T],
```

Comment:
Why not `ScafiDevice` directly?

Reply:
Partially addressed in this [commit](https://github.com/AlchemistSimulator/Alchemist/pull/1300/commits/7bab1d016458adb474632259c3d2e8ecfac54943).